src/QsCompiler/SyntaxProcessor/CapabilityInference/ResultAnalyzer.fs

```diff
@@ -42,30 +42,55 @@ let createPattern kind range =
     let capability = TargetCapability.withResultOpacity opacity TargetCapability.bottom
 
     let diagnose (target: Target) =
-        let range = QsNullable.defaultValue Range.Zero range
-
-        match kind with
-        | _ when TargetCapability.subsumes target.Capability capability -> None
-        | ConditionalEqualityInOperation
-        | UnrestrictedEquality ->
-            let code =
+        if target.UseWarnings then
+            let range = QsNullable.defaultValue Range.Zero range
+
+            match kind with
+            | _ when TargetCapability.subsumes target.Capability capability -> None
+            | ConditionalEqualityInOperation
+            | UnrestrictedEquality ->
+                let code =
+                    if target.Capability.ResultOpacity = ResultOpacity.opaque then
+                        WarningCode.UnsupportedResultComparison
+                    else
+                        WarningCode.ResultComparisonNotInOperationIf
+                Some(code, [ target.Name ])
+            | ReturnInDependentBranch ->
+                if target.Capability.ResultOpacity = ResultOpacity.opaque then
+                    None
+                else
+                    Some(WarningCode.ReturnInResultConditionedBlock, [ target.Name ])
+            | SetInDependentBranch name ->
+                if target.Capability.ResultOpacity = ResultOpacity.opaque then
+                    None
+                else
+                    Some(WarningCode.SetInResultConditionedBlock, [ name; target.Name ])
+            |> Option.map (fun (code, args) -> QsCompilerDiagnostic.Warning (code, args) range)
+
+        else
+            let range = QsNullable.defaultValue Range.Zero range
+
+            match kind with
```

Comment:
We've discussed an alternative approach with Sarah. This alternative would keep F# code unchanged generating errors unconditionally. Then it will conditionally change the severity of some errors to warnings if necessary.
Each approach has merits and drawbacks. I tried this approach in another (private) branch and I didn't like it much. First, I didn't find a good way to add tests. Second, the F# code would always generate errors so it would provide different results if used by itself. Third, if some action is taken before the translation, it would use errors rather than warnings.
So I'm in favor of creating errors or warnings conditionally from the start.